lib/config.coffee

```diff
@@ -209,18 +187,54 @@ exports.getByApplication = (application, options = {}) ->
 		return config
 
 ###*
+# @summary Get a device configuration object from an application
+# @public
+# @function
+#
+# @param {String} application - application name
+# @param {Object} [options={}] - options
+# @param {String} [options.wifiSsid] - wifi ssid
+# @param {String} [options.wifiKey] - wifi key
+#
+# @returns {Promise<Object>} device configuration
+#
+# @todo Move this to the SDK
+#
+# @example
+# deviceConfig.getByApplication 'App1',
+# 	network: 'wifi'
+# 	wifiSsid: 'foobar'
+# 	wifiKey: 'hello'
+# .then (configuration) ->
+# 	console.log(configuration)
+###
+exports.getByApplication = (application, options = {}) ->
+	Promise.join(
+		resin.models.application.getApiKey(application)
+		getByApplication(application, options)
+		(apiKey, config) ->
+			config.apiKey = apiKey
+
+			exports.validate(config)
```

Comment:
We're now validating this config twice: once in getByApplication and once again here just after adding the apiKey. Should probably drop from getByApplication.